src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3612,7 +3622,8 @@ private void AddSynthesizedExtensionImplementationsIfNecessary(MembersAndInitial
                 {
                     foreach (var member in type.GetMembers())
                     {
-                        if (member is MethodSymbol { IsImplicitlyDeclared: false } method)
+                        if (member is MethodSymbol { IsImplicitlyDeclared: false } method &&
+                            (method.IsStatic || type.ExtensionParameter is not null))
```

Comment:
I don't follow why this check was added. ExtensionParameter can be null when the extension method is static?

Reply:
> I don't follow why this check was added. ExtensionParameter can be null when the extension method is static?

It is an error scenario, but the parameter is not used in static case.

Reply:
FWIW, the way I thought of this check: the implementation method only needs an extra parameter when it is for an instance method, and without an extension parameter we can't create it.